cmake/Modules/Platform/Emscripten.cmake

```
@@ -18,7 +18,12 @@ set(CMAKE_SYSTEM_NAME Emscripten)
 set(CMAKE_SYSTEM_VERSION 1)
 
 set(CMAKE_CROSSCOMPILING TRUE)
-set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS FALSE)
+
+# shared library (side module) support
+set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
+set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-s SIDE_MODULE=1")    # instead of -shared
+set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "-s SIDE_MODULE=1")  # instead of -shared
+set(CMAKE_STRIP FALSE)  # not supported
```

We could add this to keep the current default: contrary to CMake-defaults, emscripten would then default to static targets for any `add_library` call:
```suggestion
set(CMAKE_STRIP FALSE)  # not supported
set(BUILD_SHARED_LIBS OFF)  # default to static libs, even if we allow shared ones
```

---

Do you know why CMAKE_STRIP is not supported?   Is it not supported for static linking also?   At least we should file a bug a reference the bug here.